test/test_server_task.py

```diff
@@ -270,7 +270,7 @@ def test_sync_task_no_server(comm):
     try:
         client.connect()
     except Exception as exc:  # pylint: disable=broad-except
-        assert False, f"unexpected exception: {exc}"
+        raise AssertionError(f"unexpected exception: {exc}") from exc
```

Comment:
Any thoughts on removing this extra `try-except` (and the one above)?  If there is an `Exception` raised by `client.connect()`, the test will fail regardless.

Reply:
This is needed, until the new client code is merged. There are some strange exceptions below and I wanted to avoid making a list a of exceptions.

I like your change to reraise.

Reply:
Alright, sounds good!  Then this PR is ready